### DISCUSSION TITLE
Feature request (on plugin): Include protocol errors (TIMEDOUT or CONNECTRESET) in plugin Metrics by endpoint

### DISCUSSION THREAD

**Discussion by joan-teriihoania** [artilleryio/artillery#1547] [Artillery]
I'm using the plugin [artillery-plugin-metrics-by-endpoint](https://github.com/artilleryio/artillery-plugin-metrics-by-endpoint) to get stats per endpoint which works fine for HTTP error codes (400-599) but not protocol errors (e.g. `ECONNRESET` or `ETIMEDOUT`).

My use case is that I need to get stats on an app using two types of requests with different behaviors: one asks for a token, and the other sends a file that will be loaded and parsed by the backend. I have a phase that sends a lot of files to parse which may overload the server, in which case I want to know how much and which requests got blocked.

Is there a way to do that in the plugin already? If not, would it be possible to implement it in the plugin?
Or is there another solution I'm not seeing?
Thank you very much for your help.

**Comment by hassy** [comment]
hi @joan-teriihoania 👋 protocol-level errors are tracked by Artillery itself and will be reported in the metrics output.

You can try it yourself by setting a target to a hostname that can't be resolved, e.g. `https://arandomdomainnamethatdefinitelydoesnotexist1234.com`

**Reply by joan-teriihoania** [reply]
Hi! Yes, I am aware. What I was wondering was if it was possible to have which endpoint was the origin of the error.

For example, I have a scenario which sends two requests :
 - GET /api/account (`get_profile`)
 - POST /api/account (`update_profile`)

At the moment, Artillery (with the plugin) gives a report like that:
```
http.requests: 30
http.codes.200: 10
plugins.[...].http.codes.200.get_profile: 5
plugins.[...].http.codes.200.update_profile: 5
http.codes: 404: 16
plugins.[...].http.codes.404.get_profile: 8
plugins.[...].http.codes.404.update_profile: 8
errors.ETIMEDOUT: 4
```

(At least, that's what I'm getting... Please tell me if this isn't the expected behavior or just not under the purview of the plugin)

Though the report says which endpoint (get_profile or update_profile) sent a request that got a 200 or 404 response, I don't know which got an TIMEDOUT error (though for this example, I could easily guess, I can't do that with 24000+ sample size).

Is it possible to have a report like that with the plugin?
```
http.requests: 30
http.codes.200: 10
plugins.[...].http.codes.200.get_profile: 5
plugins.[...].http.codes.200.update_profile: 5
http.codes: 404: 16
plugins.[...].http.codes.404.get_profile: 8
plugins.[...].http.codes.404.update_profile: 8
errors.ETIMEDOUT: 4
>> plugins.[...].errors.ETIMEDOUT.get_profile: 2
>> plugins.[...].errors.ETIMEDOUT.update_profile: 2
```

**Reply by hassy** [reply]
I see what you mean, thank you for the extra details!

Yes it should be possible to have the plugin report errors for a specific endpoint.
